### PR TITLE
Implemented tags

### LINKED ISSUE #362: Implement tags
User should be able to create tags and assign them to a course. Additionally, users should be able to filter courses by tags.

### PR DESCRIPTION
**Pull request by paulsojan**
Fixes #362

### CODE REVIEW

**Comment by paulsojan** [comment]
@deepakkumarnd Please review

File: app/views/shared/components/_alert.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<% submitButtonLabel ||= "OK" %>
```
Comment (deepakkumarnd) [inline app/views/shared/components/_alert.html.erb:1]: use snake case submit_button_label as variable name

File: spec/requests/tags_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,174 @@
+# frozen_string_literal: true
```
Comment (deepakkumarnd) [inline spec/requests/tags_controller_spec.rb:1]: since this is a request spec so rename the file name accordingly

**Comment by paulsojan** [comment]
@deepakkumarnd Made changes. Please review

**Pull request merged**